src/NUnitFramework/framework/nunit.framework.csproj

```diff
@@ -15,6 +15,10 @@
     <PackageReference Include="NUnit.System.Linq" Version="0.6.0" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
+    <PackageReference Include="Microsoft.Win32.Registry" Version="4.4.0" />
```

Comment:
I didn't see the registry being used. Do we need this WIndows compat pack package?

Reply:
Unfortunately, yes. That code is used to detect the .NET Framework version by looking in the registry in RuntimeFramework.cs. We need it for .NET Standard as well in case the .NET standard framework is running on .NET Framework instead of .NET Core (although, I'm not sure that would ever happen).

Reply:
Hmm...not a nice dependency for a .NET Standard build!

As far as I can see (please doubel check my logic!), in RuntimeFramework, the registry lookup is only used for determining if the current framework is any of .NET 2.0/3.0/3.5. Given the .NET Standard build will never be running on any of those platforms, could we pre-processor out the `Registry.OpenSubkey` in NET_STANDARD20, then remove this dependency? And of course throw an exception should the 'impossible' ever happen? 🙂

Reply:
Right, the .NET Standard 1.4 build shouldn't be checking for any versions of .NET Framework lower than 4.6.1.

Reply:
None of this code will be running with .NET Standard 1.4. It doesn't have the APIs available to do platform detection. I can remove the registry check from RuntimeFramework.cs where it checks for frameworks < .NET 4, which will never happen, but OSPlatform.cs still needs the registry. If we aren't removing the registry, the code will be more clear if we don't #if out that section of code (imho).

Reply:
The .NET Standard 2.0 build shouldn't be checking for any versions of .NET Framework lower than 4.6.1 either, since those frameworks can't run it.

Reply:
It shouldn't be actually checking, but itt has to return not supported for those frameworks that can't run it in order for the Platform attribute to work correctly. We used to do something like this in the CF build... we conditionally included code to reject everything except CF.

Reply:
@MatthewBeardmore So instead of adding the registry dependency, why not out `#if !NETSTANDARD2_0` around this, since there's no way .NET Framework's 2.0 CLR could be running the .NET Standard build?

https://github.com/nunit/nunit/blob/49ded54b0f68f333399b8931e6e0a9151df92ee7/src/NUnitFramework/framework/Internal/RuntimeFramework.cs#L104-L123

`throw PlatformNotSupportedException();` would be fine.

Reply:
It wouldn't be possible to get into that section, which is why I've added an #ifdef around that section. The registry is needed to detect Windows 8+ versioning if the manifest is not set up with a GUID that marks it as supporting Windows 8 and higher, which we can't remove without a difference between the .NET Standard and .NET Framework builds.

Reply:
@MatthewBeardmore You're quite right, not sure what I was thinking at the time. Thanks for changing this section!